Update release process to support release candidate (#132)
DataDog/datadog-operator: RELEASING.md

# Release process

## Steps

> 1. Checkout the repository on the correct branch and changeset (`master`)

Reviewer: ```suggestion
1. Checkout the repository on the correct branch and changeset (`master`).
```

> 2. Run pre-release script

Reviewer: ```suggestion
2. Run the pre-release script:
```

>     1. Release candidate
> 
>        In case of release candidate, run localy commmand `make VERSION=x.v.z RELEASE_CANDIDATE=1 pre-release`
> 
>        For example `make VERSION=0.3.0 RELEASE_CANDIDATE=2 pre-release` will generate the release version `0.3.0-rc.2`
> 
>     2. Final release
> 
>        Run localy the command `make VERSION=x.v.z pre-release`

Reviewer: ```suggestion
    - For a release candidate, run the following command, locally: `make VERSION=x.v.z RELEASE_CANDIDATE=1 pre-release`

      For example, to generate the release version `0.3.0-rc.2`, run: `make VERSION=0.3.0 RELEASE_CANDIDATE=2 pre-release`
     - For a final release, run the following command, locally: `make VERSION=x.v.z pre-release`
```

3. Commit all the changes generated from the previous command:

   ```console
   $ git add .

>    $ git commit -s -m "release vX.Y.X"

Reviewer 2: I think you mean `git commit -S` instead of the lowercase `-s`

Author: both work on my machine :)

Reviewer 2: They do different things. Lowercase does the "Signed-off-by", while uppercase does GPG signing. If both work for you, it might be because you have `commit.gpgsign` set to true (either on the repo, or on your global config), which might not be the case for everyone.

Author: right

   # or
   $ git commit -s -m "release vX.Y.X-rc.W"
   ```

> 4. Uncomment the job `push_latest_to_docker_hub` in `.gitlab-ci.yml` in case you plan to push the image to dockerhub with the `latest` image tag.

Reviewer 3: Shouldn't it be a manual job only accessible on a tag?

Author: Yes, I didn't change this process, it was like this before. But I can convert it as a manual job. 
I think the initial idea, was to comment it in order to avoid any mistake

Author: I have check and it is already uncommented

5. Add release tag, correct format: `git tag vX.Y.Z` or `git tag vX.Y.Z-rc.W`
6. Push the generated commit and tag to the repostory branch.

   ```console
   $ git push origin vX.Y.Z
   ```

## Other PRs to create

> - Create krew PR for the plugin on https://github.com/kubernetes-sigs/krew-index to update the `datadog.yaml` artifact. (See [kubernetes-sigs/krew-index#727](https://github.com/kubernetes-sigs/krew-index/pull/727) as an example)

Reviewer: ```suggestion
- Create krew PR for the plugin on https://github.com/kubernetes-sigs/krew-index to update the `datadog.yaml` artifact. (See [kubernetes-sigs/krew-index#727](https://github.com/kubernetes-sigs/krew-index/pull/727) as an example).
```
